utils.js

```diff
@@ -240,17 +266,23 @@ export async function id() {
 export async function delegate({zcap, signer, capabilityChain}) {
   capabilityChain =
     Array.isArray(capabilityChain) ? capabilityChain : [zcap.parentCapability];
+  let suite;
+  if(signer.type === 'Ed25519VerificationKey2018') {
+    suite = new Ed25519Signature2018({
+      signer,
+    });
+  } else if(signer.type === 'Ed25519VerificationKey2020') {
+    suite = new Ed25519Signature2020({
+      signer,
+    });
+  }
```

Comment:
I just noticed, that we should throw an error if we hit the bottom without finding a matching `signer.type`.